src/fixit/rule/cst.py

```diff
@@ -69,10 +73,26 @@ def visit_hook(name: str) -> Iterator[None]:
                 logger.debug(f"PERF: {name} took {duration_us} µs")
                 self.timings[name] += duration_us
 
+        metadata_cache: Mapping[ProviderT, object] = {}
+        needs_repo_manager: Set[ProviderT] = set()
+
         for rule in rules:
             rule._visit_hook = visit_hook
+            if FullyQualifiedNameProvider in rule.METADATA_DEPENDENCIES:
```

Comment:
This is not enough, we need to inspect transitive dependencies of the rule. You'll want to call `get_inherited_depedencies()`: https://py.wtf/libcst/libcst._metadata_dependent/MetadataDependent#libcst._metadata_dependent.MetadataDependent.get_inherited_dependencies

Reply:
sidenote: it sucks that there's no good way to inspect metadataproviders to see if they need a `FullRepoManager` (today the best way is to check if they have a `gen_cache()` classmethod)